lib/Basics/debugging.cpp

```diff
@@ -204,3 +204,8 @@ template<>
 char const conpar<false>::open = '[';
 template<>
 char const conpar<false>::close = ']';
+
+thread_local std::ostringstream
+    arangodb::debug::AssertionLogger::assertionStringStream;
+thread_local arangodb::debug::AssertionConditionalStream
+    arangodb::debug::AssertionConditionalLogger::assertionStringStream;
```

Comment:
@goedderz : Do we need to define these thread_local objects when we are not in maintainer mode?
Or rather, should we put these definitions inside an `#ifdef`?

Reply:
I think they're not necessary, I've wrapped them in an `#ifdef`. However, we should then also do this in `devel` (this is just a backport). I'll remind myself to do that next week.